service/history/replication/task_processor.go

```diff
@@ -448,13 +448,16 @@ func (p *taskProcessorImpl) processTaskOnce(replicationTask *types.ReplicationTa
 	if err != nil {
 		p.updateFailureMetric(scope, err)
 	} else {
-		domainName, errorDomainName := p.shard.GetDomainCache().GetDomainName(replicationTask.HistoryTaskV2Attributes.DomainID)
-		if errorDomainName != nil {
-			return errorDomainName
-		}
 		now := ts.Now()
 		mScope := p.metricsClient.Scope(scope, metrics.TargetClusterTag(p.sourceCluster))
-		mScope.Tagged(metrics.DomainTag(domainName))
+		domainID := replicationTask.HistoryTaskV2Attributes.GetDomainID()
+		if domainID != "" {
+			domainName, errorDomainName := p.shard.GetDomainCache().GetDomainName(domainID)
+			if errorDomainName != nil {
+				return errorDomainName
+			}
+			mScope.Tagged(metrics.DomainTag(domainName))
```

Comment:
I think this should be:
mScope = mScope.Tagged(metrics.DomainTag(domainName))

Reply:
oh right, my bad!